src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs

```diff
@@ -67,14 +67,17 @@ internal Task OnProjectFactoryCompletedAsync()
                 ruleNames: _targetFrameworkWatchedRules,
                 initialDataAsNew: false, // only reset on subsequent changes
                 suppressVersionOnlyUpdates: true);
-
             return Task.CompletedTask;
         }
 
+        public Task UnloadAsync() => DisposeAsync();
```

Comment:
No, Unload != Dispose. Unload is "temporarily shutdown". 

Easiest way to test this is to add Directory.Build.targets in the same directory and continually comment/uncomment and Save the file.

``` XML
 <ItemGroup>
    <ProjectCapability Remove="PackageReferences"/>
  </ItemGroup>
```

This should cause Load/Unload to be called over and over again.